substrate/client/network/src/litep2p/mod.rs

```diff
@@ -907,7 +907,7 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkBackend<B, H> for Litep2pNetworkBac
 								);
 
 								self.event_streams.send(Event::Dht(
-									DhtEvent::ValueNotFound(libp2p::kad::RecordKey::new(&key))
+									DhtEvent::ValueNotFound(sc_network_types::rec::Key::new::<sc_network_types::rec::Key>(&key.to_vec().into()))
```

Comment:
You are passing the same type as generic argument?

Reply:
This is because of the `.into()` call, as the compiler does not know what to convert the `key.to_vec()` into. Just keeping it as `key.to_vec()` should be enough. Also, the `KademliaKey` that is needed here is under your control, so you can implement `From::<libp2p::kad::RecordKey>` for it and use `key.into()` here instead of` sc_network_types::rec::Key::new()`.